### PR TITLE
Add `wrapping`

### PR DESCRIPTION
**Pull request by iaik-jheher**
Add a `wrapping` helper that wraps any exception as the specified type.

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -1,2 +1,2 @@
 
-rootProject.name = "kmmresult"
\ No newline at end of file
+rootProject.name = "KmmResult"
```
Comment (iaik-jheher) [inline settings.gradle.kts:2]: (needed to make the project work on non-case-sensitive OS)

**Comment by iaik-jheher** [comment]
@JesusMcCloud

File: settings.gradle.kts (hunk 1 above)
Comment (JesusMcCloud) [reply settings.gradle.kts:2]: i don't know what that does to the publication, though

**Review by JesusMcCloud — APPROVED** [review]
smooth!

**Pull request merged**